src/main/java/com/intuit/graphql/orchestrator/xtext/XtextGraphBuilder.java

```diff
@@ -4,14 +4,33 @@
 import com.intuit.graphql.orchestrator.ServiceProvider;
 import com.intuit.graphql.orchestrator.schema.Operation;
 import com.intuit.graphql.orchestrator.utils.XtextUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.util.EnumMap;
 import java.util.Map;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
 import org.eclipse.xtext.resource.XtextResourceSet;
 
+@Slf4j
 public class XtextGraphBuilder {
 
   public static XtextGraph build(ServiceProvider serviceProvider) {
 
+    if(serviceProvider.getSeviceType() == ServiceProvider.ServiceType.FEDERATION) {
+      try {
+        String federation_directives = IOUtils.toString(Files.newInputStream(Paths.get("src/main/resources/federation_built_in_directives.graphqls")));
```

Comment:
Make sure this file is loaded only once, keep it public static final.

Reply:
Added other federation specs